docs/dev/testing.md

```diff
@@ -90,32 +99,162 @@ reject file:
+10:05:52.688 INFO> started server at host 127.159.235.1 in scylla-1, pid 2165602
+```
+
+Next, we can take a look at the server log, which is at
+`/opt/local/work/scylla/scylla/testlog/dev/scylla-1.log`:
+
+The log contains special markers, written at test start and end:
+
+```
+INFO  2022-08-18 10:05:52,598 [shard 0] schema_tables - Schema version changed to 8b5e9c73-7c1c-3b28-8c31-c1359210c484
+------ Starting test test_null.1 ------
+...
+INFO  2022-08-18 10:05:53,297 [shard 0] schema_tables - Dropping keyspace cql_test_1660806353124
+INFO  2022-08-18 10:05:53,304 [shard 0] schema_tables - Schema version changed to 8b5e9c73-7c1c-3b28-8c31-c1359210c484
+------ Ending test test_null.1 ------
+```
+
+Most often there are no errors in Scylla log, so next we inspect
+the test' log, which is next to the server's at
+`/opt/local/scylla/scylla/testlog/dev/test_null.1.log`:
+```
+        cql.execute(f"INSERT INTO {table1} (p,c) VALUES ('{p}', '3')")
+>       assert False
+E       assert False
+```
+
+What does number 1 mean in the log file name? Since test.py can
+run parallel jobs and run each test multiple times with `--repeat`,
+each execution is assigned a unique sequence number, allowing to
+distinguish artifacts of different execution.
+
+When finished debugging, you don't have to worry about deleting the remains
+of a previous run, `test.py` will clean then up on the next execution
+automatically.
+
+### Pooling implementation details
+
+When pooling and running multiple servers, we want to avoid host/port or
+temporary directory clashes. We also want to make sure that `test.py`
+doesn't leave any running servers around, even when it's interrupted
+by user or with an exception. This is why `test.py` has a special
+registry where it tracks all servers, in which each server gets a unique
+address in a subnet of network `127.*.*.*`. Unless killed with `SIGKILL`,
+`test.py` kills all servers it creates at shutdown.
+
+The servers created by the pool use a pre-defined set of options
+to speed up boot. Some of these options are developer-only, such as
+`flush_schema_tables_after_modification: false`. If you wish to
+extend the options of a used server, you can do it by adding
+`extra_scylla_cmdline_options` or `extra_scylla_config_options`
+to your suite.yaml.
+
+## Topology pytests
+
+In addition to the standard 'Python' suite type, `test.py`
+supports an extended pytest suite, `Topology`. Unlike
+`Python` tests, `Topology` tests run against scylla clusters,
+and support topology operations. A standard `manager`
+fixture is available for these. Through this fixture,
+you can access individual cluster instances, start, stop
+and restart instances, add and remove instances from the cluster.
+The `manager` fixture connects to the cluster by sending
+`test.py` REST commands over a socket file http connection.
+This guarantees that `test.py` is fully aware of all topology
+operations and can clean up resources, including added
+servers, when tests end.
```

Comment:
I'd mention cluster reuse, dirty clusters, and per test case cluster health check.

Reply:
Done, cluster health checks are mentioned elsewhere.